File: Week_6/src/calculator.js
var Calculator = function () {

    this.add = function (a, b) {
        return a + b;
    }

    this.subtract = function (a, b) {
        return a - b;
    }

    this.multiply = function (a, b) {
        return a * b;
    }

    this.divide = function (a, b) {
        return a / b;
    }

    this.factorial = function (a) {
        if (a < 0) {
            throw new Error("Cannot compute a negative factorial")
        }
        if (a === 1 || a === 0) {
            return 1;
        }
        return a * this.factorial(a - 1);
    }
}

File: Week_6/Week_6_workspace/WebDriverExample/src/main/java/com/revature/stepImplementations/Calculator.java
package com.revature.stepImplementations;

public class Calculator {

	public int add(int a, int b) {
		return a + b;
	}

	public int subtract(int a, int b) {
		return a - b;
	}

	public int multiplication(int x, int y) {
		return x * y;
	}

	public int divide(int a, int b) {
		if (b==0) {
			throw new IllegalArgumentException("Denominator cannot equal 0");
		}
		return a / b;
	} 
}
